File: 2023-2024/sourcecode/mod4/lab12/RetrofitBasicsJava/app/src/main/java/com/example/retrofitbasicsjava/data/repository/PostRepository.java
package com.example.retrofitbasicsjava.data.repository;

import com.example.retrofitbasicsjava.data.Post;
import com.example.retrofitbasicsjava.data.api.PlaceholderApi;
import com.example.retrofitbasicsjava.data.api.RetrofitInstance;

import java.io.IOException;
import java.util.List;

import retrofit2.Call;

public class PostRepository {
    private final PlaceholderApi api = RetrofitInstance.getApi();

    public Call<List<Post>> getPosts() throws IOException {
        return api.posts();
    }
}

File: 2023-2024/sourcecode/mod3/RetrofitBasicsCompose/app/src/main/java/com/example/retrofitbasicscompose/data/repository/PostRepository.kt
package com.example.retrofitbasicscompose.data.repository

import com.example.retrofitbasicscompose.api.RetrofitInstance
import com.example.retrofitbasicscompose.data.Post

class PostRepository {
    private val api = RetrofitInstance.api

    suspend fun getPosts(): List<Post>{
        return api.posts()
    }
}